colorbleed/plugins/maya/load/load_yeti_cache.py

```diff
@@ -0,0 +1,166 @@
+        if not cmds.pluginInfo("pgYetiMaya", query=True, loaded=True):
+            cmds.loadPlugin("pgYetiMaya", quiet=True)
+
+        # Get JSON
+        fname, ext = os.path.splitext(self.fname)
+        settings_fname = "{}.fursettings".format(fname)
+        with open(settings_fname, "r") as fp:
+            fursettings = json.load(fp)
+
+        # Get node name from JSON
+        nodes = []
+        for node, settings in fursettings.items():
+
+            # Create new pgYetiMaya node
+            node_name = "{}:{}".format(namespace, node)
+            yeti_node = cmds.createNode("pgYetiMaya", name=node_name)
+            cmds.connectAttr("time1.outTime", "%s.currentTime" % yeti_node)
+
+            # Apply colorbleed ID to node
+            lib.set_id(node=yeti_node,
+                       unique_id=settings["cbId"],
+                       overwrite=True)
+            settings.pop("cbId", None)
+
+            # Apply settings
+            for attr, value in settings.items():
+                attribute = "%s.%s" % (yeti_node, attr)
+                cmds.setAttr(attribute, value)
+
+            # Ensure the node has no namespace identifiers
+            node = node.replace(":", "_")
+
+            # Create full cache path
+            cache = "{}/{}.%04d.fur".format(self.fname, node)
+            cache_fname = self.validate_cache(cache)
+            cache_path = os.path.join(self.fname, cache_fname)
+
+            # Add filename to `cacheFileName` attribute, set to cache mode
+            cmds.setAttr("%s.cacheFileName" % yeti_node,
+                         cache_path,
+                         type="string")
+
+            cmds.setAttr("%s.fileMode" % yeti_node, 1)
+
+            nodes.append(yeti_node)
+
+        group_name = "{}:{}".format(namespace, asset["name"])
+        group_node = cmds.group(nodes, name=group_name)
+        all_nodes = cmds.listRelatives(group_node, children=True)
+        all_nodes.append(group_node)
+
+        self[:] = all_nodes
+
+        return pipeline.containerise(name=name,
+                                     namespace=namespace,
+                                     nodes=all_nodes,
+                                     context=context,
+                                     loader=self.__class__.__name__)
+
+    def remove(self, container):
+
+        from maya import cmds
+
+        namespace = container["namespace"]
+        container_name = container["objectName"]
+
+        self.log.info("Removing '%s' from Maya.." % container["name"])
+
+        container_content = cmds.sets(container_name, query=True)
+        nodes = cmds.ls(container_content, long=True)
+
+        nodes.append(container_name)
+
+        try:
+            cmds.delete(nodes)
+        except ValueError:
+            # Already implicitly deleted by Maya upon removing reference
+            pass
+
+        cmds.namespace(removeNamespace=namespace, deleteNamespaceContent=True)
+
+    def update(self, container, representation):
```

Comment:
Note that updating logic eventually will become more complex. For example in these scenarios:

- What if a pgYetiMaya node in the source file was removed or added?
- What if the render density or render width in the source file was changed? Should this override it again? What if the local scene had "overrides"?

---

And a little note to self, the `.fur` representation is a folder, so this `api.get_representation_path(representation)` will return a folder, not a file - which is fine.